### PR TITLE
kmod: remove sof_snd_intel_client

### PR DESCRIPTION
**Pull request by fredoh9**
Need to remove snd_sof_intel_client first and snd_sof_client
before snd_sof.

Signed-off-by: Fred Oh <fred.oh@linux.intel.com>

Ancillary Bus need sof_snd_intel_client to remove before sof_snd.
https://sof-ci.01.org/linuxpr/PR2264/build4588/devicetest/

### CODE REVIEW

File: tools/kmod/sof_remove.sh [hunk 1]
```diff
@@ -55,6 +55,7 @@ remove_module snd_soc_sof_sdw
 remove_module snd_soc_ehl_rt5660
 remove_module snd_sof_ipc_test
 remove_module snd_sof_probes
+remove_module snd_sof_intel_client
```
Comment (plbossart) [inline tools/kmod/sof_remove.sh:58]: don't we also need remove_module snd_sof_client?

Comment (fredoh9) [reply tools/kmod/sof_remove.sh:58]: by Makefile change, the module name is snd_sof_intel_client

`rmmod: ERROR: Module snd_sof is in use by: snd_sof_intel_client`

Comment (plbossart) [reply tools/kmod/sof_remove.sh:58]: snd_sof_intel_client uses snd_sof_client in the latest PR, see e.g. https://github.com/thesofproject/linux/pull/2264/commits/e0333e48f760e08e95635960fe4ed4072b37e790

Comment (fredoh9) [reply tools/kmod/sof_remove.sh:58]: thanks, I tested in APL UP2. Let me check how that passed

**Comment by marc-hb** [comment]
Small Yak Shaving ON: could we "upgrade" this file and document inter-module dependencies using comments? Here's one way it could probably be done:

- First, play some long-duration sound to make sure modules are in active use. Meanwhile:
- For each module in the file, run: `lsmod | grep ' snd_sof_intel_hda_common'`.
- Take all names on the left (in this example: `snd_soc_hdac_hda`, `snd_sof_xtensa_dsp`, `snd_sof_intel_hda` and `snd_hda_ext_core` ) and add them as a comment above the `snd_sof_intel_hda_common` line in this file.

Now I admit such a documentation will never be perfect because some product-specific drivers will always be underdocumented and dependencies  could change. Nevertheless, it would still make the maintenance of this file (updates and code reviews) much simpler and faster.

File: tools/kmod/sof_remove.sh (hunk 1 above)
Comment (fredoh9) [reply tools/kmod/sof_remove.sh:58]: @plbossart I checked the Makefile again. there is no snd_sof_client module. sof-client.c is part of snd-soc.ko. I remember also it was separate module in between version, but not anymore. Do I miss anything?

Comment (ranj063) [reply tools/kmod/sof_remove.sh:58]: @plbossart the way the Makefile is set up now sof-client.c will be built into the snd-sof module. let me make a minor change to the Makefile to make it a separate module

Comment (ranj063) [reply tools/kmod/sof_remove.sh:58]: but is there a particular reason why it should be a separate module?

Comment (plbossart) [reply tools/kmod/sof_remove.sh:58]: > but is there a particular reason why it should be a separate module?

you export a namespace, it'd be odd if that namespace was part of another module which defines another namespace.

**Comment by plbossart** [comment]
> Small Yak Shaving ON: could we "upgrade" this file and document inter-module dependencies using comments? Here's one way it could probably be done:
> 
> * First, play some long-duration sound to make sure modules are in active use. Meanwhile:
> * For each module in the file, run: `lsmod | grep ' snd_sof_intel_hda_common'`.
> * Take all names on the left (in this example: `snd_soc_hdac_hda`, `snd_sof_xtensa_dsp`, `snd_sof_intel_hda` and `snd_hda_ext_core` ) and add them as a comment above the `snd_sof_intel_hda_common` line in this file.
> 
> Now I admit such a documentation will never be perfect because some product-specific drivers will always be underdocumented and dependencies could change. Nevertheless, it would still make the maintenance of this file (updates and code reviews) much simpler and faster.

The dependencies are relatively trivial: top-level, machine drivers, codec drivers, remaining sound modules.

File: tools/kmod/sof_remove.sh [hunk 2]
```diff
@@ -55,6 +55,8 @@ remove_module snd_soc_sof_sdw
 remove_module snd_soc_ehl_rt5660
 remove_module snd_sof_ipc_test
 remove_module snd_sof_probes
+remove_module snd_sof_client
+remove_module snd_sof_intel_client
```
Comment (ranj063) [inline tools/kmod/sof_remove.sh:59]: should it be intel_client first and then snd_sof_client?

**Review by plbossart — APPROVED** [review]
thanks @fredoh9

**Comment by marc-hb** [comment]
> The dependencies are relatively trivial: top-level, machine drivers, codec drivers, remaining sound modules.

Great, even simpler. @fredoh9 can you please add these 4 section headers as comments in the file?

**Review by ranj063 — APPROVED** [review]

File: tools/kmod/sof_remove.sh [hunk 3]
```diff
@@ -58,6 +65,9 @@ remove_module snd_sof_probes
 remove_module snd_sof_intel_client
 remove_module snd_sof_client
 
+#-------------------------------------------
+# Codec driver
+#-------------------------------------------
```
Comment (plbossart) [inline tools/kmod/sof_remove.sh:70]: this should be two lines down. This part  should be 'Remaining core SOF parts"

File: tools/kmod/sof_remove.sh [hunk 4]
```diff
@@ -108,6 +118,9 @@ remove_module snd_soc_dmic
 remove_module snd_hda_codec_realtek
 remove_module snd_hda_codec_generic
 
+#-------------------------------------------
+# Remaning sound modules
```
Comment (plbossart) [inline tools/kmod/sof_remove.sh:122]: typo: Remaining

**Review by plbossart — CHANGES_REQUESTED** [review]
Minor changes in comments needed. thanks!

File: tools/kmod/sof_remove.sh (hunk 3 above)
Comment (fredoh9) [reply tools/kmod/sof_remove.sh]: I moved down by two lines,

I believe this is applicable to below section(last block)
>This part should be 'Remaining core SOF parts"

**Review by plbossart — APPROVED** [review]

**Review by marc-hb — APPROVED** [review]

**Pull request merged**